datacube/model/__init__.py

```diff
@@ -504,6 +535,86 @@ def extract_point(name):
 
         return GridSpec(crs=crs, **gs_params)
 
+    def validate_extra_dims(definition: dict):
+        """Validate extra_dimension metadata in the product definition.
+
+        :param definition: Dimension definition dict, typically retrieved from the product definition's
+            `extra_dimensions` field.
```

Comment:
A brief description of what this is doing would be nice. What is this code checking, what failure modes it is supposed to detect.

Reply:
Will do.

Reply:
Done, Please let me know if there are any other required changes.